### PR TITLE
br: support backup and restore `column_stats_usage`

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/53567

Problem Summary:

After we enabled the predicate columns feature by default, we need to support backup and restore this information.

### What changed and how does it work?

- Added the support for backuping and restoring `column_stats_usage`.
- I moved some functions from `pkg/statistics/handle/usage/predicate_column.go` to `pkg/statistics/handle/usage/predicatecolumn/predicate_column.go`, otherwise it would cause a cyclic dependence issue.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] [Manual test](https://github.com/pingcap/tidb/pull/54634#issuecomment-2232896962)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
支持备份和恢复 mysql.column_stats_usage 表
Support for backing up and restoring mysql.column_stats_usage table
```

### CODE REVIEW

File: pkg/statistics/handle/usage/predicatecolumn/predicate_column.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	return err
+}
+
+// SaveColumnStatsUsageForTable saves column stats usage information for a specific table to disk.
+func SaveColumnStatsUsageForTable(
+	sctx sessionctx.Context,
+	colStatsUsage map[model.TableItemID]statstypes.ColStatsTimeInfo,
+) error {
+	for colID, statsUsage := range colStatsUsage {
+		lastUsedAt := "NULL"
```
Comment (hi-rustin) [inline pkg/statistics/handle/usage/predicatecolumn/predicate_column.go:142]: I am not sure if we have a better way to handle this. The ExecRows function cannot handle nil correctly, so I have to rely on the CONVERT_TZ to insert 'null' if it is 'NULL'.

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Comment by hi-rustin** [comment]
Tests locally:
1. Start the TiDB cluster with patch: `tiup playground nightly --db.binpath /Volumes/t7/code/tidb/bin/tidb-server`
2. Build the BR: `make build_br`
3. Backup the cluster to a local file path: `br backup full --pd "127.0.0.1:2379" --storage local:///Volumes/t7/code/brtest`
4. Restore from the local file: `br restore db --pd "127.0.0.1:2379" --db test --storage local:///Volumes/t7/code/brtest`

**Review by hawkingrei — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Pull request merged**